mbed-client-libservice/platform/arm_hal_interrupt.h

```diff
@@ -30,6 +30,19 @@ extern void platform_enter_critical(void);
  */
 extern void platform_exit_critical(void);
 
+/**
+ * \brief This function increments the disable IRQ counter.
+ * The function must be called inside interrupt routines
+ * after platform_enter_critical() has been called.
+ */
+extern void platform_entered_interrupt(void);
+/**
+ * \brief This function decrements the disable IRQ counter.
+ * The function must be called inside interrupt routines
+ * after platform_exit_critical() has been called.
+ */
+extern void platform_exit_interrupt(void);
```

Comment:
exiting

Not sure you need to mention exit_critical() here - this just balances the entered_interrupt call.